src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -208,6 +212,19 @@ def _accumulate_by_children(self, area, flattened, results):
                 )
         return results
 
+    def _aggregate_totals_with_penalties(self, area, results):
+        if not area.children:
+            return
+        totals_with_penalties = 0
+        for child in area.children:
+            self._aggregate_totals_with_penalties(child, results)
+            if "totals_with_penalties" in results[child.name]:
+                if isinstance(results[child.name]["totals_with_penalties"], dict):
```

Comment:
Can you tell me why `results[child.name]["totals_with_penalties"]` is expected to be in two different data type ?

Reply:
The reason for this is that is that it `totals_with_penalties` in an non-device area would be also a number. but as all members of an area are subdicts (e.g. `{"grid": {"House 1": {}, "House 2":{}, "Accumulated Trades": {}, "External Tradds": {}}}` )i followed this convention and created a subdict for the `totals_with_penalties` too and put it in a subdict.
As we foresie some additional parameters that need to be accumulated like this, a subdict is helpful too. 
IN the devices `totals_with_penalties` is a normal entry.
Does it make sense?

Reply:
Thanks for the explanation, makes sense now BUT better add an additional check to validate its not a dict in any case if it is a device

Reply:
Hmm , i dont get it. This IS the check whether it is the device or not, by checking if it is a dict or not.

Reply:
This is an implicit way of checking it and will work until devices' `totals_with_penalties` are not accidentally populated with dict due to some logical failure.

So, better a unittest validating that `_store_area_penalties` method will never populate a dict type `totals_with_penalties` for devices